org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/IProjectsManager.java

```diff
@@ -103,4 +103,30 @@ default void unregisterListeners() {
 	default void projectsImported(IProgressMonitor monitor) {
 		// do nothing
 	}
+
+	/**
+	 * Executed after the projects are imported and finished.
+	 */
+	default void projectsBuildFinished(IProgressMonitor monitor) {
+		// do nothing
+	}
+
+	/**
+	 * Check whether the build is finished
+	 */
+	default boolean isBuildFinished() {
+		return true;
+	};
+
+	default boolean shouldUpdateProjects() {
+		return false;
+	}
+
+	default void setShouldUpdateProjects(boolean update) {
+		// do nothing
+	}
+
+	default void updateMavenProjects(IProgressMonitor monitor) {
```

Comment:
I don't think this should be part of the API. Is there any way to call it something like `updateProjects` and just pass in some kind of filter that gets the correct projects ?

Reply:
I will try.